### PR TITLE
LPS-197960 [PRM] Create retry logic for Object Proxy Salesforce requests

### PR DESCRIPTION
**Pull request by thaynam**

### CODE REVIEW

**Comment by thaynam** [comment]
ci:test:sf

**Comment by thaynam** [comment]
ci:test:sf

**Comment by thaynam** [comment]
ci:test:relevant

**Comment by thaynam** [comment]
ci:test:stable

**Comment by thaynam** [comment]
ci:forward:force

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/PartnerOpportunitiesList/hooks/useGetListItemsFromPartnerOpportunities.tsx [hunk 1]
```diff
@@ -6,25 +6,22 @@
...
+import useGet from '../../../common/services/liferay/object/useGet';
 import getOpportunityAmount from '../utils/getOpportunityAmount';
 import getOpportunityDates from '../utils/getOpportunityDates';
 
 export default function useGetListItemsFromPartnerOpportunities(
 	page: number,
 	pageSize: number,
 	filtersTerm: string,
-	sort: string,
-	opportunityFilter?: string
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/PartnerOpportunitiesList/hooks/useGetListItemsFromPartnerOpportunities.tsx:19]: Why `opportunityFilter` was removed?

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useDynamicFieldEntries.ts [hunk 2]
```diff
@@ -6,15 +6,29 @@
...
+
+	const {data: listTypeDefinitions} = useGet<
+		LiferayItems<ListTypeDefinition[]>
+	>(
+		`/o/${
+			LiferayAPIs.HEADERLESS_ADMIN_LIST_TYPE
+		}/list-type-definitions?filter=name in ('${pickListNames.join(
+			"', '"
+		)}')`
+	);
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFRequestList/hooks/useDynamicFieldEntries.ts:31]: Join with one element? It's better to use the element directly.
```suggestion
	const {data: listTypeDefinitions} = useGet<
		LiferayItems<ListTypeDefinition[]>
	>(
		`/o/${
			LiferayAPIs.HEADERLESS_ADMIN_LIST_TYPE
		}/list-type-definitions?filter=name eq ${LiferayPicklistName.MDF_REQUEST_STATUS}`
	);
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useDynamicFieldEntries.ts [hunk 3]
```diff
@@ -6,15 +6,29 @@
...
+
+	const {data: listTypeDefinitions} = useGet<
+		LiferayItems<ListTypeDefinition[]>
+	>(
+		`/o/${
+			LiferayAPIs.HEADERLESS_ADMIN_LIST_TYPE
+		}/list-type-definitions?filter=name in ('${pickListNames.join(
+			"', '"
+		)}')`
+	);
```
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/MDFClaimList/hooks/useDynamicFieldEntries.ts:31]:
```suggestion
	const {data: listTypeDefinitions} = useGet<
		LiferayItems<ListTypeDefinition[]>
	>(
		`/o/${
			LiferayAPIs.HEADERLESS_ADMIN_LIST_TYPE
		}/list-type-definitions?filter=name eq ${LiferayPicklistName.MDF_CLAIM_STATUS}`
	);
```

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/utils/retry.ts
Comment (jairmedeiros) [inline workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/common/utils/retry.ts:1]: The retry.ts logic need to be the same of useGet with retry.

**Review by jairmedeiros — CHANGES_REQUESTED** [review]

File: workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/PartnerOpportunitiesList/hooks/useGetListItemsFromPartnerOpportunities.tsx (hunk 1 above)
Comment (thaynam) [reply workspaces/liferay-partner-workspace/client-extensions/liferay-partner-custom-element/src/routes/PartnerOpportunitiesList/hooks/useGetListItemsFromPartnerOpportunities.tsx:19]: I changed it to `renewalOpportunitiesFilter`

**Pull request closed without merging**